executors/statusHandler.go

```diff
@@ -11,26 +11,26 @@ import (
 	"github.com/multiversx/mx-chain-keys-monitor-go/core"
 )
 
+const statusHandlerName = "statusHandler"
+
 type statusHandler struct {
-	name            string
-	mut             sync.Mutex
-	numErrors       uint32
-	problematicKeys map[string]struct{}
-	notifiers       []OutputNotifier
+	name             string
+	mut              sync.Mutex
+	numErrors        uint32
+	problematicKeys  map[string]struct{}
+	notifiersHandler OutputNotifiersHandler
 }
 
 // NewStatusHandler creates a new instance of type statusHandler
-func NewStatusHandler(name string, notifiers []OutputNotifier) (*statusHandler, error) {
-	for index, notifier := range notifiers {
-		if check.IfNil(notifier) {
-			return nil, fmt.Errorf("%w at index %d", errNilOutputNotifier, index)
-		}
+func NewStatusHandler(name string, notifiersHandler OutputNotifiersHandler) (*statusHandler, error) {
```

Comment:
add a check for the `name` parameter to not be empty ?

Reply:
it can be empty, it is mostly used on logs